SHARE-458 Redesign of comment section

dermaku-dardan opened the pull request:
The comment section needed some aesthetic changes and the option to reply to a comment. Changes regarding to the appearance of the comments were realised and now the users can also reply to a comment

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [ ] Include the name and id of the Jira ticket in the PR’s title eg.: `SHARE-666 The devils ticket`
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no warnings and errors 
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Verify that all tests are passing (CI), if not please state the test cases in the [section](#Tests) below
- [ ] Perform a self-review of the changes
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [ ] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer
- [ ] Check that your pull request has been successfully deployed to https://web-test-1.catrob.at/

### Additional Description
`TODO: Add additional information that is not in your commit-message here`

### Tests - additional information
`TODO: add additional information about testruns here`

dmetzner commented on translations/catroweb.en_AU.yaml:
```diff
@@ -207,6 +207,9 @@ programs:
   createTranslation: Create translation
+  deletedComment: '**Deleted**'
+  replyLabel: Reply
+  repliesLabel: Replies
```
you only have to change the base file
catroweb.en.yml.

All other files will be generated with Crowdin.

dmetzner commented on assets/js/custom/ProgramComments.js:
```diff
@@ -16,19 +16,38 @@ export function ProgramComments (programId, visibleComments, showStep, minAmount
     updateButtonVisibility()
   })
 
+  $(function () {
```
please don't use jquery, we are trying to get rid of it

dmetzner commented on assets/js/custom/ProgramComments.js:
```diff
@@ -72,10 +93,13 @@ export function ProgramComments (programId, visibleComments, showStep, minAmount
     }
+    
+    var postCommentUrl = $('.js-project-comments').data('path-post-comment-url')
+    var parentCommentId = $('.js-project-parentComment').data('parent-comment-id')
```
```suggestion
    const postCommentUrl = $('.js-project-comments').data('path-post-comment-url')
    const parentCommentId = $('.js-project-parentComment').data('parent-comment-id')
```

dmetzner commented on assets/js/custom/ProgramComments.js:
```diff
@@ -93,8 +117,10 @@ export function ProgramComments (programId, visibleComments, showStep, minAmount
 
   function deleteComment (commentId) {
+    const $projectComments = $('.js-project-comments')
+    var deleteCommentUrl = $projectComments.data('path-delete-comment-url')
```
```suggestion
    const deleteCommentUrl = $projectComments.data('path-delete-comment-url')
```

dmetzner commented on assets/js/custom/ProgramComments.js:
```diff
@@ -179,12 +207,26 @@ export function ProgramComments (programId, visibleComments, showStep, minAmount
 
   function restoreAmountOfVisibleCommentsFromSession () {
-    const lastSessionAmount = JSON.parse(window.sessionStorage.getItem('visibleComments'))
+    const lastSessionAmount = getVissibleCommentsSessionVar()
```
```suggestion
    const lastSessionAmount = getVisibleCommentsSessionVar()
```

dmetzner commented on assets/js/custom/ProgramComments.js:
```diff
@@ -219,15 +263,31 @@ export function ProgramComments (programId, visibleComments, showStep, minAmount
+
+  function getVissibleCommentsSessionVar() {
+    const visibleCommentSessionVarName = getVissibleCommentsSessionVarName()
+    return JSON.parse(window.sessionStorage.getItem(visibleCommentSessionVarName, JSON.stringify(amountOfVisibleComments)))
```
visible typo

dmetzner commented on assets/js/custom/TranslateComments.js:
```diff
@@ -9,20 +9,23 @@ export class TranslateComments extends Translation {
+    $('.comment-translation-button').on('click', function (event) {
+      event.stopPropagation()
+      const commentId = $(this).attr('id').substring('comment-translation-button-'.length) 
+      var translateCommentUrl = $('.comment-translation').data('path-translate-comment')
```
```suggestion
      const translateCommentUrl = $('.comment-translation').data('path-translate-comment')
```

dmetzner commented on assets/styles/custom/program.scss:
```diff
@@ -242,23 +245,30 @@ table td:nth-child(1) {
+// .name-date-wrapper>* {
+//   margin: 30px 0;
+// }
+
```
can be removed :)

dmetzner commented on migrations/2022/Version20220916214048.php:
```diff
@@ -0,0 +1,39 @@
+<?php
```
Please use a single migration for every ticket / feature and remove the faulty code snippets

Pretty sure you haven't changed studio - that's a bug in the generation with enums :/

dermaku-dardan committed:
SHARE-458 Redesign of comment section

The comment section needed some aesthetic changes and the option to reply to a comment. Changes regarding to the appearance of the comments were realised and now the users can also reply to a comment.

dmetzner committed:
Fix entity manager is closed bug and static analysis issued of PR SHARE-458

dmetzner committed:
Fix test using russian translation

dmetzner committed:
Fix tests for SHARE-458
